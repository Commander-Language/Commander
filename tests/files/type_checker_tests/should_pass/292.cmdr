//Test type statements type check
type output = (string, string, int);
type outputs = output[];
x: outputs = [`echo 'hi'`, `echo 'bye'`];
