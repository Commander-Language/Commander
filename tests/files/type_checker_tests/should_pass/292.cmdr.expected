(PrgmNode
	(TypeStmtNode output (TupleTypeNode (TypesNode (StringTypeNode STRING) (StringTypeNode STRING) (IntTypeNode INT)) (STRING, STRING, INT)))
	(TypeStmtNode outputs (ArrayTypeNode (VariableTypeNode output (STRING, STRING, INT)) (STRING, STRING, INT)[]))
	(ExprStmtNode (BinOpExprNode (BindingNode x (VariableTypeNode outputs (STRING, STRING, INT)[])) SET (ArrayExprNode (ExprsNode (CmdExprNode (BasicCmdNode (StringNode 'echo') (StringNode (StringExprsNode (StringExpressionNode (StringNode 'hi') STRING)))) (STRING, STRING, INT)) (CmdExprNode (BasicCmdNode (StringNode 'echo') (StringNode (StringExprsNode (StringExpressionNode (StringNode 'bye') STRING)))) (STRING, STRING, INT))) (STRING, STRING, INT)[]) (STRING, STRING, INT)[])))
